python/ray/data/dataset.py

```diff
@@ -4647,6 +4662,30 @@ class MaterializedDatastream(Datastream, Generic[T]):
     pass
 
 
+@PublicAPI(stability="beta")
+class Schema:
+    """Datastream schema.
+
+    Attributes:
+        names: List of column names of this Datastream.
+        base_schema: The underlying Arrow or Pandas schema.
+    """
+
+    def __init__(self, base_schema: Union["pyarrow.lib.Schema", "PandasBlockSchema"]):
+        self.base_schema = base_schema
+
+    @property
+    def names(self) -> List[str]:
+        """Lists the columns of this Datastream."""
+        return self.base_schema.names
+
+    def __str__(self):
+        return f"Schema({str(self.names)})"
```

Comment:
Can we make the repr `Schema({"col1": type1, "col2": type2, "col3": type3", ...})` instead? That would not only be much more useful (include the type), but also have the same format as the dataset rows themselves (a dict of column name to type) :)

Reply:
Let me look into this. One blocker is consistently mapping arrow to pandas types, but this sounds fixable at least for the string form.